packages/plugin/src/theia.d.ts

```diff
@@ -7571,6 +7581,13 @@ declare module '@theia/plugin' {
          * may list our every specific kind they provide, such as `CodeActionKind.Refactor.Extract.append('function`)`
          */
         readonly providedCodeActionKinds?: ReadonlyArray<CodeActionKind>;
+
+        /**
+         * Documentation from the provider is shown in the code actions menu
+         *
+         * At most one documentation entry will be shown per provider.
+         */
+        documentation?: ReadonlyArray<{ command: Command, kind: CodeActionKind }>
```

Comment:
I don't see how the documentation field is transferred to the main side and ultimately the monaco editor.

Reply:
I think we can omit support for `documentation` for now, it is used when registering `registerCodeActionsProvider` but our implementation differs from vscode in many ways:

https://github.com/eclipse-theia/theia/blob/0f7dec1acc4eb738111ed33d41bb5bd47e9bf897/packages/plugin-ext/src/plugin/languages.ts#L441-L456

- https://github.com/microsoft/vscode/blob/abfaf30a04ffd25f9b3a835479379329ab947495/src/vs/workbench/api/common/extHostLanguageFeatures.ts#L1785-L1797

Reply:
But ultimately, we're registering the code action provider to the editor, which _does_ have a documentation field. We should pass this info to the editor in Theia, as well.

Reply:
@tsmaeder

Reply:
Our definition of modes.CodeActionProvider is _way_ out of date. We need to update our definition in monaco.d.ts and pass the necessary information to the front end through "registerQuickFixProvider". The current definition in monaco actually looks like this: https://github.com/microsoft/vscode/blob/2586299c42490a8ed6dff96ed731d44fcbe5b20e/src/vs/editor/common/languages.ts#L777

Reply:
I looks like in the branch we currently use (0.23) of vs code, the definition lives here: (https://github.com/microsoft/vscode/blob/b882940dc6d5b2acb096969b9e43385248719df7/src/vs/editor/common/modes.ts#L740). 

I believe it would work to extend the definition of CodeActionProvider in `monaco.d.ts` with the needed bits.

Reply:
@Archie27376 : Please have a look at the hints that Thomas provided and let us know if you need further information!

Reply:
@JonasHelming Dante's (@Archie27376) internship has terminated, someone else would need to pick up the pull-request.